### PR TITLE
Task/1.4.1 upgrade

### PR DESCRIPTION
**Pull request by shapirov103**
*Issue #, if available:*

*Description of changes:* Tested pipeline with the stable node/npm versions, updated README and performance improvements.

`make build` is a pre-requisite to running `cdk` commands.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,10 +29,10 @@ Update (provided Node version manager is installed): `n stable`. May require `su
 
 ```bash
 $ npm -v
-8.7.0
+8.19.2
 ```
 
-Updating npm: `npm install -g npm@latest` where latest can also be a specific version above 8.4. May require `sudo`.
+Updating npm: `sudo n stable` where stable can also be a specific version above 8.4. May require `sudo`.
```
Comment (youngjeong46) [inline README.md:35]: wouldn't this require module `n`?

**Review by youngjeong46 — COMMENTED** [review]
One minor comment, other wise LGTM.

File: README.md (hunk 1 above)
Comment (youngjeong46) [reply README.md:35]: Disregard. I did not see the update note above. LGTM.

Comment (shapirov103) [reply README.md:35]: you can test it out. npm install will have n installed. I have this code in the codebuild step of the pipeline and I tested it out on my branch.

**Review by youngjeong46 — APPROVED** [review]
LGTM

**Pull request merged**